File: apps/events-service/src/controllers/health/health.controller.ts
import { Controller, Get } from '@nestjs/common';
import { PrismaService } from '../../prisma.service';

@Controller('health')
export class HealthController {
  constructor(private readonly prisma: PrismaService) {}

  @Get()
  async getHealth() {
    try {
      await this.prisma.$queryRaw`SELECT 1`;

      return {
        status: 'healthy',
        service: 'events-service',
        timestamp: new Date().toISOString(),
        database: 'connected',
      };
    } catch (error) {
      console.log('Health check error:', error);

      return {
        status: 'unhealthy',
        service: 'events-service',
        timestamp: new Date().toISOString(),
        database: 'disconnected',
        error: 'Internal server error',
      };
    }
  }
}

File: apps/events-service-springboot/src/main/java/com/alexaxenti/events_service_springboot/health/HealthController.java
package com.alexaxenti.events_service_springboot.health;

import org.springframework.jdbc.core.JdbcTemplate;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.time.Instant;
import java.util.LinkedHashMap;
import java.util.Map;

@RestController
public class HealthController {
    private final JdbcTemplate jdbcTemplate;

    public HealthController(JdbcTemplate jdbcTemplate) {
        this.jdbcTemplate = jdbcTemplate;
    }

    @GetMapping("/health")
    public Map<String, Object> getHealth() {
        Map<String, Object> response = new LinkedHashMap<>();

        try {
            jdbcTemplate.queryForObject("SELECT 1", Integer.class);
            response.put("status", "healthy");
            response.put("service", "events-service");
            response.put("timestamp", Instant.now().toString());
            response.put("database", "connected");
        } catch (Exception exception) {
            response.put("status", "unhealthy");
            response.put("service", "events-service");
            response.put("timestamp", Instant.now().toString());
            response.put("database", "disconnected");
            response.put("error", "Internal server error");
        }

        return response;
    }
}
